### PR TITLE
[WIP] Implement new Baggage APIs

### PR DESCRIPTION
**Pull request by slashmo**
This PR incorporates the changes introduced in [Baggage Context](https://github.com/slashmo/gsoc-swift-baggage-context) 0.5.0.

### CODE REVIEW

File: Package.resolved [hunk 1]
```diff
@@ -2,12 +2,12 @@
   "object": {
```
Comment (ktoso) [inline Package.resolved]: might remove this file early btw, we'll want to not have it eventually

File: Sources/Tracing/NoOpTracer.swift [hunk 2]
```diff
@@ -12,23 +12,24 @@
...
 import Baggage
+import BaggageContext
 import Instrumentation
 
 /// No operation Tracer, used when no tracing is required.
 public struct NoOpTracer: Tracer {
     public func startSpan(
         named operationName: String,
-        context: BaggageContextCarrier,
+        context: Context,
```
Comment (ktoso) [inline Sources/Tracing/NoOpTracer.swift]: not `BaggageContext`?

File: Sources/Tracing/NoOpTracer.swift [hunk 3]
```diff
@@ -12,23 +12,24 @@
...
         ofKind kind: SpanKind,
         at timestamp: Timestamp
     ) -> Span {
-        return NoOpSpan(context: context.baggage)
+        return NoOpSpan(baggage: context.baggage)
```
Comment (ktoso) [inline Sources/Tracing/NoOpTracer.swift]: +1

File: Sources/Tracing/Span.swift [hunk 4]
```diff
@@ -11,16 +11,16 @@
...
 /// A `Span` type that follows the OpenTracing/OpenTelemetry spec. The span itself should not be
 /// initializable via its public interface. `Span` creation should instead go through `tracer.startSpan`
 /// where `tracer` conforms to `Tracer`.
 ///
 /// - SeeAlso: [OpenTelemetry Specification: Span](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#span).
 public protocol Span: AnyObject {
-    /// The read-only `BaggageContext` of this `Span`, set when starting this `Span`.
-    var context: BaggageContext { get }
+    /// The read-only `Baggage` of this `Span`, set when starting this `Span`.
+    var baggage: Baggage { get }
```
Comment (ktoso) [inline Sources/Tracing/Span.swift:23]: 👍

File: Sources/Tracing/Tracer.swift [hunk 5]
```diff
@@ -17,15 +17,16 @@ import Instrumentation
...
+    ///   - baggage: The `Baggage` providing information on where to start the new `Span`.
     ///   - kind: The `SpanKind` of the new `Span`.
     ///   - timestamp: The `DispatchTime` at which to start the new `Span`.
     func startSpan(
         named operationName: String,
-        context: BaggageContextCarrier,
+        baggage: Baggage,
         ofKind kind: SpanKind,
         at timestamp: Timestamp
     ) -> Span
```
Comment (ktoso) [inline Sources/Tracing/Tracer.swift:32]: Good, and we might want an overload accepting `BaggageContext` 👍

**Review by ktoso — APPROVED** [review]
Looking good! Thanks for the work on migrating over.
Happy to merge and keep working on it in smaller commits soon

File: Sources/Tracing/Tracer.swift (hunk 5 above)
Comment (slashmo) [reply Sources/Tracing/Tracer.swift:32]: Good idea. I'll create an issue for it.

**Pull request merged**